### ISSUE TITLE
take use_alter=True, other FK / foreign key / foreignkey constraint rules into account when autogenerating create_table / drop_table

### ISSUE THREAD

**Issue by sqlalchemy-bot** [sqlalchemy/alembic#326]
**Migrated issue, originally created by Adrian Vogelsgesang ([@vogelsgesang](https://github.com/vogelsgesang))**

If the schema contains circular dependencies, the order of the `CREATE TABLE` statements is incorrect.

This is because `metadata.sorted_tables` (used [here](https://bitbucket.org/zzzeek/alembic/src/1c67b1ab8c6fe273d4e175a14f0df5d3cbbd0edc/alembic/autogenerate/compare.py?at=master&fileviewer=file-view-default#compare.py-73)) returns the tables ordered alphabetically if there are circular dependencies.

**Comment by zzzeek** [comment]
@CaselIT this is kind of a pretty big hole in autogenerate

**Comment by zzzeek** [comment]
yeah.   we know how to determine that using what is now called [sort_tables_and_constraints](https://github.com/sqlalchemy/sqlalchemy/blob/805a1323b973a30af99ce506dd5c5c4ab96cff0f/lib/sqlalchemy/sql/ddl.py#L1257) so this would need to be run and rearrange the tables somewhere between the compare and render phases.  like perhaps sort and rearrange when we have the migration structure built.

**Comment by adamsaimi** [comment]
Hello, this issue is relevant to our work organization. I will try to tackle this in an upcoming merge request using the sqlalchemy [sort_tables_and_constraints](https://github.com/sqlalchemy/sqlalchemy/blob/805a1323b973a30af99ce506dd5c5c4ab96cff0f/lib/sqlalchemy/sql/ddl.py#L1257)

**Comment by paulius2k** [comment]
Hello, are there still hopes that this will be resolved anytime? 
Thanks.
